TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/tracing/IReplayContexts.java

```diff
@@ -0,0 +1,347 @@
+package org.opensearch.migrations.replay.tracing;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.AttributesBuilder;
+import org.opensearch.migrations.replay.datatypes.ISourceTrafficChannelKey;
+import org.opensearch.migrations.replay.datatypes.ITrafficStreamKey;
+import org.opensearch.migrations.replay.datatypes.UniqueReplayerRequestKey;
+import org.opensearch.migrations.tracing.IScopedInstrumentationAttributes;
+import org.opensearch.migrations.tracing.IWithTypedEnclosingScope;
+
+import java.time.Instant;
+
+public abstract class IReplayContexts {
+
+    public static class ActivityNames {
```

Comment:
Would this be a lot more efficient as a protobuf defined enum? Reading the [OTEL Spec](https://opentelemetry.io/docs/specs/otlp/#json-protobuf-encoding), it seems they have made optimizations to require integer serialization for enums which is stricter than the standard protobuf spec.

Reply:
Let me know if you think that is just referring to the otel structure itself as opposed to the span attributes

Reply:
I'm not sure that it is relevant for us, but knowing how protobuf works, they'd need to be creating on the fly schema definitions anyway & I don't think that it would matter if these were strings (that could be mapped) or ints (which could reach into the billions).

There might be a lot of value in doing this as an enum.  I plan to pull MetricNames and ActivityNames at some point into tests to confirm that they're all mapping to observed traces and metrics within tests (effectively more coverage stats).